src/main/java/org/junit/rules/TemporaryFolder.java

```diff
@@ -170,56 +170,51 @@ public File newFile() throws IOException {
      * Returns a new fresh folder with the given path under the temporary
      * folder.
      */
-    public File newFolder(String folder) throws IOException {
-        if (new File(folder).isAbsolute()) {
-            throw new IOException("folder name must be a relative path");
-        }
-        File file = new File(getRoot(), folder);
-        if (!file.mkdirs()) {
-            throw new IOException(
-                    "a folder with the name \'" + folder + "\' already exists");
-        }
-        return file;
+    public File newFolder(String path) throws IOException {
+        return newFolder(new String[]{path});
     }
 
     /**
-     * Returns a new fresh folder with the given name(s) under the temporary
+     * Returns a new fresh folder with the given paths under the temporary
      * folder. For example, if you pass in the strings {@code "parent"} and {@code "child"}
      * then a directory named {@code "parent"} will be created under the temporary folder
      * and a directory named {@code "child"} will be created under the newly-created
      * {@code "parent"} directory.
      */
-    public File newFolder(String... folderNames) throws IOException {
-        File file = getRoot();
-        for (int i = 0; i < folderNames.length; i++) {
-            String folderName = folderNames[i];
-            validateFolderName(folderName);
-            file = new File(file, folderName);
-            if (!file.mkdir() && isLastElementInArray(i, folderNames)) {
+    public File newFolder(String... paths) throws IOException {
+        if (paths.length == 0) {
+            throw new IllegalArgumentException("must pass at least one path");
+        }
+
+        /*
+         * Before checking if the paths are absolute paths, check if create() was ever called,
+         * and if it wasn't, throw IllegalStateException.
+         */
```

Comment:
I think the comment is wrong because I could not find the `IllegalStateException`.